src/components/MovableCamera.cxx

```diff
@@ -0,0 +1,72 @@
+/*
+ *  MovableCamera.cxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <iostream>
+#include <Urho3D/Input/Input.h>
+#include <Urho3D/UI/UI.h>
+#include "components/MovableCamera.hxx"
+
+MovableCamera::MovableCamera(Urho3D::Context* context) : Urho3D::Camera{context}, key_map{} {
+    key_map[forward_key] = Urho3D::Vector3::FORWARD;
+    key_map[back_key] = Urho3D::Vector3::BACK;
+    key_map[left_key] = Urho3D::Vector3::LEFT;
+    key_map[right_key] = Urho3D::Vector3::RIGHT;
+}
+
+void MovableCamera::RegisterObject(Urho3D::Context* context) { context->RegisterFactory<MovableCamera>(); }
+
+void MovableCamera::setSpecialKey(specialKey special_key, int new_key) {
+    auto reset_key = [&](int& key, const Urho3D::Vector3& vec) {
+        unsetKey(key);
+        key = new_key;
+        setKey(key, vec);
+    };
+    switch (special_key) {
+    case FORWARD:
+        reset_key(forward_key, Urho3D::Vector3::FORWARD);
+        break;
+    case BACK:
+        reset_key(back_key, Urho3D::Vector3::BACK);
+        break;
+    case LEFT:
+        reset_key(left_key, Urho3D::Vector3::LEFT);
+        break;
+    case RIGHT:
+        reset_key(right_key, Urho3D::Vector3::RIGHT);
+        break;
+    default:
+        break;
+    }
```

Comment:
This could go away with a simple array:
Along the lines of
```cpp
using Urho3D::Vector3;
std::array vecs {{Vector3::FORWARD, Vector3::BACK, Vector3::LEFT, Vector3::RIGHT}};
reset_key(user_keys[static_cast<int>(special_key)], vecs[static_cast<int>(special_key)]);
```